### ISSUE TITLE
Potential Improvement to the value training target

### ISSUE THREAD

**Issue by vishvananda** [tensorflow/minigo#296]
Hello,

My team had a discussion with one of the authors of minigo at a booth at the nVidia GTC. I can't remember his/your name, but we discussed one potential improvement we had made to the AlphaZero algorithm regarding the training target for the value output. We've finally had a chance to write a detailed explanation of the improvement in a blog post here: https://medium.com/oracledevs/lessons-from-alphazero-part-4-improving-the-training-target-6efba2e71628

We haven't tried the approach on a high branching-factor game like go, and we are very curious if you find a similar improvement using our alternative target. Let me know if that is interesting and if I can help.

**Comment by sethtroisi** [comment]
Hey Vish,

People have experimented with this before and in the past it has generated worse results.

I will edit this section as I find more references:

GCP mentioning Aya linear interpolation of Result based on move number [1]



[1] https://groups.google.com/d/msg/computer-go-archive/grF3_hdk66U/T42bZODRAQAJ
[1a] http://computer-go.org/pipermail/computer-go/2017-January/009830.html

**Comment by amj** [comment]
Hi Vish!  You spoke with me at Nvidia GTC -- thanks for looking us up!

I think it's a pretty interesting idea, for sure, and it seems to have some analogues with the way we use a softmax of the mcts results as our policy target.

I have a couple concerns, though:
1. A good value function seems to be the *most important part* of a good go engine.  Without a good value estimate, a good policy function can kinda-sorta play go, but it has no real meaning underneath.  
2. the result of the game, *z*, is the *only* true piece of information about the game that is fed as input to train the network.  the policy distribution is not 'empirically true' in the way that the result of the game is an actual truth.
3. As a result, it seems like training on q alone runs the risk of creating highly re-enforcing behaviors -- i.e., it thinks black wins, so white resigns, so the next iteration also thinks black wins, etc.  

I'm intrigued at the idea of combining z and q in some way; for instance, ignoring q if the sign is wrong, and combining them in some proportion if they agreed.  It seems worth some experiments at least :)

love the blog posts, keep 'em coming!

**Comment by amj** [comment]
preventing 'incorrect' resigns is covered in the original paper; the basic idea is to have a holdout set of games with resignation disabled, and calculate the appropriate resignation threshold accordingly.

Resignation is kind of unique to Go, because how a go game "ends" is weird.  Basically, both players have to agree that a game is over for it to be over, compared to chess or other games that have clear, unambiguous terminal conditions.  This means that a functioning tree search should never pass if it's losing -- any move should be have a higher chance of winning than agreeing to lose, basically :)  

In practice, with tromp taylor scoring, this often means that a winning player will start passing, until the losing player encircles enough clear territory to be 'winning' under TT rules, at which point the losing player starts passing and the winning player starts removing its captured stones, etc.  This is the "right" thing for MCTS to do, but it's a super, super inefficient use of computing resources, dragging games out hundreds of moves until the losing player has no alternatives other than pass.

One idea that @lightvector suggested was adjusting z by the known false-positive ratio; so if you know 3% of your resignations are incorrect, training to 0.97 instead of 1 for resigned games.
